packages/@aws-cdk/aws-apigateway/lib/deployment.ts

```diff
@@ -133,7 +133,31 @@ class LatestDeploymentResource extends CfnDeployment {
     });
 
     this.api = props.restApi;
-    this.originalLogicalId = Stack.of(this).getLogicalId(this);
+
+    const originalLogicalId = Stack.of(this).getLogicalId(this);
+
+    this.overrideLogicalId(Lazy.stringValue({ produce: ctx => {
+      const hash = [ ...this.hashComponents ];
+
+      if (this.api instanceof RestApi || this.api instanceof SpecRestApi) { // Ignore IRestApi that are imported
+
+        // Add CfnRestApi to the logical id so a new deployment is triggered when any of its properties change.
+        const cfnRestApiCF = (this.api.node.defaultChild as any)._toCloudFormation();
+        hash.push(ctx.resolve(cfnRestApiCF));
```

Comment:
Did you intentionally just use the current scope to resolve instead of the stack? I know the current scope wasn't available before in the `prepare()`, just making sure. 

```suggestion
        hash.push(Stack.of(ctx).resolve(cfnRestApiCF));
```

Also, can we replace `ctx` with `scope`? Makes it clearer.

Reply:
This is actually _not_ the scope. It's what we call "resolution context" and actually the "right way" to resolve tokens in lazy functions.

The original idea was that this would be the only way to resolve tokens, to enforce that people don't resolve too early (for example, resolve without dependencies being processed!!).

Practically it's the same resolve as `stack.resolve()` but not the same function.

Reply:
That makes sense, thinking about which scope this was did get me a bit puzzled. I'm still a little iffy with this part of the code :)